File: stock_ai_trading/app/models/model_management.py
# -*- coding: utf-8 -*-
"""
Model Management SQLAlchemy models (cleaned ASCII version).
This file removes corrupted docstrings/comments and preserves schema used by APIs.
"""

from datetime import datetime, date
from decimal import Decimal
from typing import Optional

from sqlalchemy import (
    Column, String, Integer, BigInteger, DECIMAL, Text, JSON,
    DateTime, Date, Boolean, Enum, ForeignKey, Index, UniqueConstraint, func
)
from sqlalchemy.ext.declarative import declarative_base
from sqlalchemy.orm import relationship
from enum import Enum as PyEnum

Base = declarative_base()

# =============================
# Enums
# =============================

class ModelType(PyEnum):
    DeepSeek = "DeepSeek"
    ChatGPT = "ChatGPT"
    Claude = "Claude"
    Llama = "Llama"
    Custom = "Custom"
    Ensemble = "Ensemble"

class ModelStatus(PyEnum):
    ACTIVE = "active"
    INACTIVE = "inactive"
    TRAINING = "training"
    ERROR = "error"
    MAINTENANCE = "maintenance"

class HealthStatus(PyEnum):
    HEALTHY = "healthy"
    WARNING = "warning"
    ERROR = "error"
    UNKNOWN = "unknown"

class WeightStrategy(PyEnum):
    EQUAL_WEIGHT = "equal_weight"
    ACCURACY_WEIGHT = "accuracy_weight"
    MANUAL_WEIGHT = "manual_weight"
    DYNAMIC_WEIGHT = "dynamic_weight"
    PERFORMANCE_WEIGHT = "performance_weight"

class VotingMethod(PyEnum):
    MAJORITY = "majority"
    WEIGHTED = "weighted"
    CONFIDENCE = "confidence"
    THRESHOLD = "threshold"

class TestType(PyEnum):
    UNIT = "unit"
    INTEGRATION = "integration"
    PERFORMANCE = "performance"
    ACCURACY = "accuracy"
    STRESS = "stress"

class TimePeriod(PyEnum):
    HOURLY = "hourly"
    DAILY = "daily"
    WEEKLY = "weekly"
    MONTHLY = "monthly"

# =============================
# Core Tables
# =============================

class AIModel(Base):
    __tablename__ = 'ai_models'

    # Basic info
    model_id = Column(String(50), primary_key=True)
    name = Column(String(100), nullable=False, unique=True)
    display_name = Column(String(150))
    description = Column(Text)

    # Type and provider
    model_type = Column(Enum(ModelType), nullable=False)
    provider = Column(String(50), nullable=False)
    model_version = Column(String(50))

    # API config
    api_key_encrypted = Column(Text)
    base_url = Column(String(500))
    api_endpoint = Column(String(200))

    # Parameters
    max_tokens = Column(Integer, default=4096)
    temperature = Column(DECIMAL(3, 2), default=Decimal('0.70'))
    top_p = Column(DECIMAL(3, 2), default=Decimal('1.00'))
    frequency_penalty = Column(DECIMAL(3, 2), default=Decimal('0.00'))
    presence_penalty = Column(DECIMAL(3, 2), default=Decimal('0.00'))

    # Metrics
    accuracy_rate = Column(DECIMAL(5, 4), default=Decimal('0.0000'))
    weight = Column(DECIMAL(5, 4), default=Decimal('1.0000'))
    avg_response_time = Column(Integer, default=0)
    success_rate = Column(DECIMAL(5, 4), default=Decimal('0.0000'))

    # Status
    # 直接使用字符串枚举值，匹配数据库中的小写枚举值
    status = Column(
        Enum('active', 'inactive', 'training', 'error', 'maintenance', name='model_status_enum'),
        nullable=False,
        default='inactive'
    )
    enabled = Column(Boolean, nullable=False, default=False)
    health_status = Column(
        Enum(
            HealthStatus,
            values_callable=lambda enum: [e.value for e in enum],
            name='healthstatus'
        ),
        default=HealthStatus.UNKNOWN
    )

    # Usage stats
    total_requests = Column(BigInteger, default=0)
    total_tokens_used = Column(BigInteger, default=0)
    last_used_at = Column(DateTime)

    # Training info
    last_trained_at = Column(DateTime)
    training_data_version = Column(String(50))
    training_status = Column(Enum('not_trained', 'training', 'completed', 'failed', name='training_status_enum'), default='not_trained')

    # Ext config
    config_json = Column(JSON)
    tags = Column(JSON)

    # Audit
    created_by = Column(String(50))
    updated_by = Column(String(50))
    created_at = Column(DateTime, nullable=False, default=func.now())
    updated_at = Column(DateTime, nullable=False, default=func.now(), onupdate=func.now())

    # Relationships
    api_keys = relationship("ModelAPIKey", back_populates="model", cascade="all, delete-orphan")
    test_records = relationship("ModelTestRecord", back_populates="model", cascade="all, delete-orphan")
    metrics = relationship("ModelMetric", back_populates="model", cascade="all, delete-orphan")
    usage_logs = relationship("ModelUsageLog", back_populates="model", cascade="all, delete-orphan")
    ensemble_mappings = relationship("EnsembleModelMapping", back_populates="model", cascade="all, delete-orphan")

    __table_args__ = (
        Index('idx_models_type', 'model_type'),
        Index('idx_models_provider', 'provider'),
        Index('idx_models_status', 'status'),
        Index('idx_models_enabled', 'enabled'),
        Index('idx_models_created_at', 'created_at'),
        Index('idx_models_accuracy', 'accuracy_rate'),
        Index('idx_models_last_used', 'last_used_at'),
    )


class ModelEnsemble(Base):
    __tablename__ = 'model_ensembles'

    ensemble_id = Column(String(50), primary_key=True)
    name = Column(String(100), nullable=False, unique=True)
    display_name = Column(String(150))
    description = Column(Text)

    weight_strategy = Column(
        Enum(WeightStrategy, values_callable=lambda enum: [e.value for e in enum], name='weightstrategy'),
        nullable=False,
        default=WeightStrategy.EQUAL_WEIGHT
    )
    voting_method = Column(
        Enum(VotingMethod, values_callable=lambda enum: [e.value for e in enum], name='votingmethod'),
        default=VotingMethod.WEIGHTED
    )
    confidence_threshold = Column(DECIMAL(3, 2), default=Decimal('0.50'))

    overall_accuracy = Column(DECIMAL(5, 4), default=Decimal('0.0000'))
    model_count = Column(Integer, default=0)
    active_model_count = Column(Integer, default=0)
    avg_response_time = Column(Integer, default=0)

    status = Column(Enum('active', 'inactive', 'configuring', 'error', name='ensemble_status_enum'), nullable=False, default='inactive')
    enabled = Column(Boolean, nullable=False, default=False)
    health_status = Column(
        Enum(HealthStatus, values_callable=lambda enum: [e.value for e in enum], name='healthstatus'),
        default=HealthStatus.UNKNOWN
    )

    total_requests = Column(BigInteger, default=0)
    success_requests = Column(BigInteger, default=0)
    last_used_at = Column(DateTime)

    config_json = Column(JSON)
    tags = Column(JSON)

    created_by = Column(String(50))
    updated_by = Column(String(50))
    created_at = Column(DateTime, nullable=False, default=func.now())
    updated_at = Column(DateTime, nullable=False, default=func.now(), onupdate=func.now())

    model_mappings = relationship("EnsembleModelMapping", back_populates="ensemble", cascade="all, delete-orphan")
    test_records = relationship("ModelTestRecord", back_populates="ensemble", cascade="all, delete-orphan")
    metrics = relationship("ModelMetric", back_populates="ensemble", cascade="all, delete-orphan")
    usage_logs = relationship("ModelUsageLog", back_populates="ensemble", cascade="all, delete-orphan")

    __table_args__ = (
        Index('idx_ensembles_strategy', 'weight_strategy'),
        Index('idx_ensembles_status', 'status'),
        Index('idx_ensembles_enabled', 'enabled'),
        Index('idx_ensembles_accuracy', 'overall_accuracy'),
        Index('idx_ensembles_created_at', 'created_at'),
    )


class EnsembleModelMapping(Base):
    __tablename__ = 'ensemble_model_mapping'

    mapping_id = Column(String(50), primary_key=True)
    ensemble_id = Column(String(50), ForeignKey('model_ensembles.ensemble_id', ondelete='CASCADE'), nullable=False)
    model_id = Column(String(50), ForeignKey('ai_models.model_id', ondelete='CASCADE'), nullable=False)

    weight = Column(DECIMAL(5, 4), nullable=False, default=Decimal('1.0000'))
    priority = Column(Integer, default=1)
    order_index = Column(Integer, default=0)

    enabled = Column(Boolean, nullable=False, default=True)
    status = Column(Enum('active', 'inactive', 'error', name='mapping_status_enum'), default='active')

    contribution_score = Column(DECIMAL(5, 4), default=Decimal('0.0000'))
    usage_count = Column(BigInteger, default=0)
    last_used_at = Column(DateTime)

    config_json = Column(JSON)

    created_by = Column(String(50))
    updated_by = Column(String(50))
    created_at = Column(DateTime, nullable=False, default=func.now())
    updated_at = Column(DateTime, nullable=False, default=func.now(), onupdate=func.now())

    ensemble = relationship("ModelEnsemble", back_populates="model_mappings")
    model = relationship("AIModel", back_populates="ensemble_mappings")

    __table_args__ = (
        UniqueConstraint('ensemble_id', 'model_id', name='uk_ensemble_model'),
        Index('idx_mapping_ensemble', 'ensemble_id'),
        Index('idx_mapping_model', 'model_id'),
        Index('idx_mapping_weight', 'weight'),
        Index('idx_mapping_priority', 'priority'),
        Index('idx_mapping_enabled', 'enabled'),
    )


class ModelTestRecord(Base):
    __tablename__ = 'model_test_records'

    test_id = Column(String(50), primary_key=True)
    model_id = Column(String(50), ForeignKey('ai_models.model_id', ondelete='CASCADE'), nullable=False)
    ensemble_id = Column(String(50), ForeignKey('model_ensembles.ensemble_id', ondelete='CASCADE'))

    # 统一枚举值存储为小写字符串
    test_type = Column(
        Enum(TestType, values_callable=lambda enum: [e.value for e in enum], name='testtype')
        , nullable=False
    )
    test_name = Column(String(200))
    test_description = Column(Text)

    input_data = Column(JSON, nullable=False)
    expected_output = Column(JSON)
    actual_output = Column(JSON)

    response_time_ms = Column(Integer)
    token_count = Column(Integer)
    success = Column(Boolean, nullable=False, default=False)
    accuracy_score = Column(DECIMAL(5, 4))

    error_code = Column(String(50))
    error_message = Column(Text)
    error_details = Column(JSON)

    test_environment = Column(String(100))
    test_version = Column(String(50))
    batch_id = Column(String(50))

    tested_by = Column(String(50))
    created_at = Column(DateTime, nullable=False, default=func.now())

    model = relationship("AIModel", back_populates="test_records")
    ensemble = relationship("ModelEnsemble", back_populates="test_records")

    __table_args__ = (
        Index('idx_test_model', 'model_id'),
        Index('idx_test_ensemble', 'ensemble_id'),
        Index('idx_test_type', 'test_type'),
        Index('idx_test_success', 'success'),
        Index('idx_test_created_at', 'created_at'),
        Index('idx_test_batch', 'batch_id'),
        Index('idx_test_response_time', 'response_time_ms'),
    )


class ModelMetric(Base):
    __tablename__ = 'model_metrics'

    metric_id = Column(String(50), primary_key=True)
    model_id = Column(String(50), ForeignKey('ai_models.model_id', ondelete='CASCADE'), nullable=False)
    ensemble_id = Column(String(50), ForeignKey('model_ensembles.ensemble_id', ondelete='CASCADE'))

    metric_date = Column(Date, nullable=False)
    metric_hour = Column(Integer)
    time_period = Column(
        Enum(TimePeriod, values_callable=lambda enum: [e.value for e in enum], name='timeperiod'),
        nullable=False,
        default=TimePeriod.DAILY
    )

    accuracy_rate = Column(DECIMAL(5, 4), default=Decimal('0.0000'))
    avg_response_time_ms = Column(Integer, default=0)
    success_rate = Column(DECIMAL(5, 4), default=Decimal('0.0000'))
    error_rate = Column(DECIMAL(5, 4), default=Decimal('0.0000'))

    total_requests = Column(BigInteger, default=0)
    successful_requests = Column(BigInteger, default=0)
    failed_requests = Column(BigInteger, default=0)
    total_tokens_used = Column(BigInteger, default=0)

    response_time_p50 = Column(Integer)
    response_time_p90 = Column(Integer)
    response_time_p95 = Column(Integer)
    response_time_p99 = Column(Integer)

    estimated_cost = Column(DECIMAL(10, 4), default=Decimal('0.0000'))
    cost_per_request = Column(DECIMAL(8, 6), default=Decimal('0.000000'))

    confidence_score = Column(DECIMAL(5, 4))
    consistency_score = Column(DECIMAL(5, 4))
    relevance_score = Column(DECIMAL(5, 4))

    custom_metrics = Column(JSON)

    created_at = Column(DateTime, nullable=False, default=func.now())
    updated_at = Column(DateTime, nullable=False, default=func.now(), onupdate=func.now())

    model = relationship("AIModel", back_populates="metrics")
    ensemble = relationship("ModelEnsemble", back_populates="metrics")

    __table_args__ = (
        UniqueConstraint('model_id', 'metric_date', 'time_period', 'metric_hour', name='uk_metrics_model_date_period'),
        Index('idx_metrics_model', 'model_id'),
        Index('idx_metrics_ensemble', 'ensemble_id'),
        Index('idx_metrics_date', 'metric_date'),
        Index('idx_metrics_period', 'time_period'),
        Index('idx_metrics_accuracy', 'accuracy_rate'),
        Index('idx_metrics_response_time', 'avg_response_time_ms'),
        Index('idx_metrics_success_rate', 'success_rate'),
        Index('idx_metrics_created_at', 'created_at'),
    )


class ModelAPIKey(Base):
    __tablename__ = 'model_api_keys'

    key_id = Column(String(50), primary_key=True)
    model_id = Column(String(50), ForeignKey('ai_models.model_id', ondelete='CASCADE'), nullable=False)

    key_name = Column(String(100), nullable=False)
    encrypted_key = Column(Text, nullable=False)
    key_hash = Column(String(64), nullable=False)
    encryption_method = Column(String(50), default='AES-256-GCM')

    rate_limit_per_minute = Column(Integer)
    rate_limit_per_day = Column(Integer)
    monthly_quota = Column(BigInteger)
    used_quota = Column(BigInteger, default=0)

    status = Column(Enum('active', 'inactive', 'expired', 'revoked', name='key_status_enum'), nullable=False, default='active')
    expires_at = Column(DateTime)
    last_used_at = Column(DateTime)

    created_by = Column(String(50))
    created_at = Column(DateTime, nullable=False, default=func.now())
    updated_at = Column(DateTime, nullable=False, default=func.now(), onupdate=func.now())

    model = relationship("AIModel", back_populates="api_keys")

    __table_args__ = (
        UniqueConstraint('model_id', 'key_name', name='uk_api_keys_model_name'),
        Index('idx_api_keys_model', 'model_id'),
        Index('idx_api_keys_status', 'status'),
        Index('idx_api_keys_expires', 'expires_at'),
    )


class ModelUsageLog(Base):
    __tablename__ = 'model_usage_logs'

    log_id = Column(String(50), primary_key=True)
    model_id = Column(String(50), ForeignKey('ai_models.model_id', ondelete='CASCADE'), nullable=False)
    ensemble_id = Column(String(50), ForeignKey('model_ensembles.ensemble_id', ondelete='CASCADE'))

    request_id = Column(String(50))
    user_id = Column(String(50))
    session_id = Column(String(100))

    input_tokens = Column(Integer, default=0)
    output_tokens = Column(Integer, default=0)
    total_tokens = Column(Integer, default=0)
    response_time_ms = Column(Integer)

    success = Column(Boolean, nullable=False, default=False)
    error_code = Column(String(50))
    error_message = Column(Text)

    estimated_cost = Column(DECIMAL(8, 6), default=Decimal('0.000000'))

    created_at = Column(DateTime, nullable=False, default=func.now())

    model = relationship("AIModel", back_populates="usage_logs")
    ensemble = relationship("ModelEnsemble", back_populates="usage_logs")

    __table_args__ = (
        Index('idx_usage_logs_model', 'model_id'),
        Index('idx_usage_logs_ensemble', 'ensemble_id'),
        Index('idx_usage_logs_user', 'user_id'),
        Index('idx_usage_logs_success', 'success'),
        Index('idx_usage_logs_created_at', 'created_at'),
        Index('idx_usage_logs_request', 'request_id'),
    )

File: stock-ai-frontend/src/api/model-management.ts
import request from '@/utils/request'

// 模型类型枚举
export enum ModelType {
  DEEPSEEK = 'DeepSeek',
  OPENAI = 'ChatGPT', 
  CLAUDE = 'Claude',
  LLAMA = 'Llama',
  CUSTOM = 'Custom'
}

// 权重策略枚举
export enum WeightStrategy {
  EQUAL_WEIGHT = 'equal_weight',
  ACCURACY_WEIGHT = 'accuracy_weight',
  MANUAL_WEIGHT = 'manual_weight',
  DYNAMIC_WEIGHT = 'dynamic_weight',
  PERFORMANCE_WEIGHT = 'performance_weight'
}

// 投票方法枚举
export enum VotingMethod {
  MAJORITY = 'majority',
  WEIGHTED = 'weighted',
  CONFIDENCE = 'confidence',
  THRESHOLD = 'threshold'
}

// 融合策略枚举（保留用于前端UI，但后端使用VotingMethod）
export enum FusionStrategy {
  WEIGHTED_VOTING = 'weighted',
  MAJORITY_VOTING = 'majority',
  CONFIDENCE_BASED = 'confidence',
  THRESHOLD_BASED = 'threshold'
}

// 模型状态枚举
export enum ModelStatus {
  ACTIVE = 'active',
  INACTIVE = 'inactive',
  TRAINING = 'training',
  ERROR = 'error'
}

// 模型接口类型定义
export interface AIModel {
  model_id: string
  name: string
  model_type: ModelType
  provider: string
  display_name?: string
  description?: string
  model_version?: string
  base_url: string
  api_key_encrypted?: string
  max_tokens?: number
  temperature?: number
  timeout?: number
  enabled: boolean
  status: ModelStatus
  accuracy?: number
  response_time?: number
  usage_count?: number
  created_at?: string
  updated_at?: string
  last_trained?: string
  config_json?: Record<string, any>
}

// 创建模型请求
export interface CreateModelRequest {
  name: string
  model_type: ModelType
  provider: string
  display_name?: string
  description?: string
  model_version?: string
  base_url: string
  api_key: string
  max_tokens?: number
  temperature?: number
  timeout?: number
  config_json?: Record<string, any>
}

// 更新模型请求
export interface UpdateModelRequest {
  name?: string
  display_name?: string
  description?: string
  model_version?: string
  base_url?: string
  api_key?: string
  max_tokens?: number
  temperature?: number
  timeout?: number
  config_json?: Record<string, any>
}

// 模型组合接口类型
export interface ModelEnsemble {
  ensemble_id: string
  name: string
  display_name?: string
  description?: string
  weight_strategy: WeightStrategy
  fusion_strategy?: FusionStrategy
  voting_method?: string
  confidence_threshold?: number
  enabled: boolean
  model_count?: number
  accuracy?: number
  created_at?: string
  updated_at?: string
  config_json?: Record<string, any>
}

// 创建组合请求
export interface CreateEnsembleRequest {
  name: string
  display_name?: string
  description?: string
  weight_strategy: WeightStrategy
  fusion_strategy?: FusionStrategy
  voting_method?: string
  confidence_threshold?: number
  config_json?: Record<string, any>
}

// 组合模型映射
export interface EnsembleModelMapping {
  ensemble_id: string
  model_id: string
  weight: number
  priority?: number
  enabled: boolean
}

// 测试结果
export interface TestResult {
  test_id: string
  model_id?: string
  ensemble_id?: string
  test_name?: string
  test_description?: string
  success: boolean
  content?: string
  output?: string
  response_time: number
  response_time_ms?: number
  accuracy_score?: number
  similarity_score?: number
  token_count?: number
  input_tokens?: number
  output_tokens?: number
  error_message?: string
  error_code?: string
  timestamp: string
  created_at?: string
  updated_at?: string
  tested_by?: string
  input_data?: {
    input?: string
    [key: string]: any
  }
  actual_output?: {
    output?: string
    [key: string]: any
  }
  expected_output?: {
    output?: string
    [key: string]: any
  }
}

// 性能统计
export interface PerformanceStats {
  total_models: number
  active_models: number
  avg_accuracy: number
  avg_response_time: number
  total_requests: number
  success_rate: number
}

// API响应基础类型
export interface ApiResponse<T = any> {
  success: boolean
  message: string
  data?: T
  error_code?: string
}

// 分页响应
export interface PaginatedResponse<T> {
  items: T[]
  total: number
  page: number
  per_page: number
  pages: number
}

// 模型管理API类
export class ModelManagementAPI {
  // 使用全局 axios baseURL('/api')，避免重复前缀导致 '/api/api'
  private baseURL = ''

  // 模型管理接口
  async getModels(params?: {
    page?: number
    per_page?: number
    search?: string
    model_type?: ModelType
    status?: ModelStatus
  }): Promise<ApiResponse<PaginatedResponse<AIModel>>> {
    return request.get(`${this.baseURL}/models`, { params })
  }

  async getModel(modelId: string): Promise<ApiResponse<AIModel>> {
    return request.get(`${this.baseURL}/models/${modelId}`)
  }

  async createModel(data: CreateModelRequest): Promise<ApiResponse<AIModel>> {
    return request.post(`${this.baseURL}/models`, data)
  }

  async updateModel(modelId: string, data: UpdateModelRequest): Promise<ApiResponse<AIModel>> {
    return request.put(`${this.baseURL}/models/${modelId}`, data)
  }

  async deleteModel(modelId: string): Promise<ApiResponse<void>> {
    return request.delete(`${this.baseURL}/models/${modelId}`)
  }

  async toggleModelStatus(modelId: string, enabled: boolean): Promise<ApiResponse<AIModel>> {
    return request.patch(`${this.baseURL}/models/${modelId}/status`, { enabled })
  }

  // 模型组合接口
  async getEnsembles(params?: {
    page?: number
    per_page?: number
    search?: string
  }): Promise<ApiResponse<PaginatedResponse<ModelEnsemble>>> {
    return request.get(`${this.baseURL}/ensembles`, { params })
  }

  async getEnsemble(ensembleId: string): Promise<ApiResponse<ModelEnsemble>> {
    return request.get(`${this.baseURL}/ensembles/${ensembleId}`)
  }

  async createEnsemble(data: CreateEnsembleRequest): Promise<ApiResponse<ModelEnsemble>> {
    return request.post(`${this.baseURL}/ensembles`, data)
  }

  async updateEnsemble(ensembleId: string, data: Partial<CreateEnsembleRequest>): Promise<ApiResponse<ModelEnsemble>> {
    return request.put(`${this.baseURL}/ensembles/${ensembleId}`, data)
  }

  async deleteEnsemble(ensembleId: string): Promise<ApiResponse<void>> {
    return request.delete(`${this.baseURL}/ensembles/${ensembleId}`)
  }

  // 组合模型管理
  async getEnsembleModels(ensembleId: string): Promise<ApiResponse<EnsembleModelMapping[]>> {
    return request.get(`${this.baseURL}/ensembles/${ensembleId}/models`)
  }

  async addModelToEnsemble(ensembleId: string, data: {
    model_id: string
    weight: number
    priority?: number
  }): Promise<ApiResponse<EnsembleModelMapping>> {
    return request.post(`${this.baseURL}/ensembles/${ensembleId}/models`, data)
  }

  async removeModelFromEnsemble(ensembleId: string, modelId: string): Promise<ApiResponse<void>> {
    return request.delete(`${this.baseURL}/ensembles/${ensembleId}/models/${modelId}`)
  }

  async updateEnsembleModelWeight(ensembleId: string, modelId: string, weight: number): Promise<ApiResponse<EnsembleModelMapping>> {
    return request.patch(`${this.baseURL}/ensembles/${ensembleId}/models/${modelId}`, { weight })
  }

  // 模型测试接口
  async testSingleModel(data: {
    model_id: string
    input: string
    expected_output?: string
    test_name?: string
    test_description?: string
  }): Promise<ApiResponse<TestResult>> {
    return request.post(`${this.baseURL}/models/test`, data)
  }

  async testEnsemble(data: {
    ensemble_id: string
    input: string
    expected_output?: string
    test_name?: string
    test_description?: string
  }): Promise<ApiResponse<TestResult>> {
    return request.post(`${this.baseURL}/ensembles/test`, data)
  }

  async getTestRecords(modelId: string, params?: {
    page?: number
    per_page?: number
    test_type?: string
    success?: boolean
  }): Promise<ApiResponse<PaginatedResponse<TestResult>>> {
    return request.get(`${this.baseURL}/models/${modelId}/test-records`, { params })
  }

  // 仪表盘统计接口
  async getDashboardStats(): Promise<ApiResponse<PerformanceStats>> {
    return request.get(`${this.baseURL}/dashboard/stats`)
  }

  async getPerformanceMetrics(params?: {
    days?: number
    model_ids?: string[]
  }): Promise<ApiResponse<any>> {
    return request.get(`${this.baseURL}/dashboard/performance`, { params })
  }

  // 健康检查
  async healthCheck(): Promise<ApiResponse<any>> {
    return request.get(`${this.baseURL}/models/health`)
  }
}

// 导出API实例
export const modelManagementAPI = new ModelManagementAPI()
